akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/JoinConfigCompatCheckerClusterSharding.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed
+
+import scala.collection.{ immutable ⇒ im }
+
+import akka.annotation.InternalApi
+import akka.cluster.{ ConfigValidation, JoinConfigCompatChecker }
+import com.typesafe.config.Config
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+final class JoinConfigCompatCheckClusterSharding extends JoinConfigCompatChecker {
+
+  override def requiredKeys: im.Seq[String] =
+    im.Seq("akka.cluster.sharding.number-of-shards")
+
+  override def check(toCheck: Config, actualConfig: Config): ConfigValidation =
+    JoinConfigCompatChecker.fullMatch(requiredKeys, toCheck, actualConfig)
```

Comment:
I created a separate issue for supporting rolling updates: https://github.com/akka/akka/issues/26194. It's not critical for this check since we have already released a few Akka versions without the check in place so users can update to those first.

Reply:
@patriknw I originally remembered that use case then forgot, thanks. Adding.. and taking on that ticket.